rust/src/address.rs

```diff
@@ -154,7 +189,7 @@ impl ByronAddress {
     /// returns the byron protocol magic embedded in the address, or mainnet id if none is present
     /// note: for bech32 addresses, you need to use network_id instead
     pub fn byron_protocol_magic(&self) -> u32 {
-        let mainnet_network_id = 764824073;
+        let mainnet_network_id = NetworkInfo::mainnet().protocol_magic();
```

Comment:
It's probably fine as-is as it's isolated in the implementation of the function that only the lib's developers will see, but maybe it'd be better to not call it network_id since we now have the distinction between network id and network/protocol magic?